Initial Classes

mecha opened the pull request.

mecha committed:
Installed dependencies

mecha committed:
Removed test templates

mecha committed:
Fixed autoload namespace

mecha committed:
Added test for `BuildWpQueryCompareCapableTrait`

mecha committed:
Added `BuildWpQueryCompareCapableTrait`

mecha committed:
Added test for `BuildWpQueryMetaCompareCapableTrait`

mecha committed:
Added `BuildWpQueryMetaCompareCapableTrait`

mecha committed:
Added test for `BuildWpQueryTaxCompareCapableTrait`

mecha committed:
Added `BuildWpQueryTaxCompareCapableTrait`

mecha committed:
Added test for `BuildWpQueryRelationCapableTrait`

mecha committed:
Added `BuildWpQueryRelationCapableTrait`

mecha committed:
Fixed code standards ...

Invoked cs-fixer on sources and tests

mecha committed:
Added PHPStorm config for PHPUnit

mecha committed:
Fixed array equivalence assertions in tests ...

Some tests were passing when you should not be, since the keys were being ignored.
These tests were amended with the proper expected result and the canonicalization was removed.

mecha committed:
Added test for `BuildWpQueryArgsCapableTrait`

mecha committed:
Added `BuildWpQueryArgsCapableTrait`

mecha committed:
Amended exception message

mecha committed:
Added missing PHP docs

mecha committed:
Fixed test name

mecha committed:
Expression support method no longer abstract ...

The `BuildWpQueryArgsCapableTrait#_isWpQueryExpressionSupported()` method is not implemented and not longer abstract.
The implementation restricts expressions by type, of which only AND types are supported.
This mirrors `WP_Query` usage, as db records must satisfy all top-level terms in the args array.

mecha committed:
Updated dependencies

mecha committed:
Compare expr support method no longer abstract ...

`BuildWpQueryCompareCapableTrait#_isWpQueryCompareExpressionSupported()` is no longer abstract.
Implementation checks if expression is an "equal to" relational type.

mecha committed:
Fixed non-absolute FQNs in test docs

mecha committed:
Added mode arg to relation building method ...

The abstract `_buildWpQueryRelationTerm()` is called for every term in an expression.
Since meta and tax queries are built differently, this abstract method will most likely need to know which is being built and use different built methods.
Therefore, the mode argument was added to provide this indication.

mecha committed:
Add test for `BuildWpQueryRelationTermCapableTrait`

mecha committed:
Added `BuildWpQueryRelationTermCapableTrait`

mecha committed:
Added `GetWpQueryRelationOperatorCapableTrait`

mecha committed:
Added `GetWpQueryMetaCompareOperatorCapableTrait`

mecha committed:
Added `GetWpQueryTaxCompareOperatorTrait`

mecha committed:
Added `GetWpQueryMetaCompareTypeCapableTrait`

mecha committed:
Renamed package with "abstract" suffix

mecha committed:
Updated readme with technical descriptions

mecha committed:
Fixed anchors in readme

mecha committed:
Improved readability of readme

- Changed trait headings to level 4
- Added eye-friendly alternative titles for easier skimming

mecha committed:
Added unfulfilled methods in readme

mecha committed:
Elaborated on unfulfilled method recommendations

mecha committed:
Fixed typos in readme

These traits respect each other very much.
But they are not in any way trying to seduce each other through compliments.
Rather, they complement each other.

XedinUnknown commented on README.md:
```diff
@@ -1,8 +1,186 @@
+## Details
+
+This package provides abstract functionality for the most implementation aspects of building [`WP_Query`] arguments from
+expressions. The traits in this package are meant to compliment each other, while also remaining agnostic of the each
```
"compliment" -> "complement"

mecha committed:
Fixed typo in readme

mecha committed:
Now throwing `OutOfRangeException` when the ...

... given expression cannot be built, either because it's not supported or its terms are incorrect.

mecha committed:
Fixed namespace case

XedinUnknown committed:
Now building also on PHP 7.2

XedinUnknown committed:
Added test folders to `autoload-dev`

XedinUnknown committed:
`develop` now aliased to `0.1.x-dev`

XedinUnknown committed:
Added changelog entry

XedinUnknown committed:
Fixed URLs in readme

XedinUnknown committed:
0Dependencies updated
